lib/src/json/reader/reader.dart

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:typed_data";
+
+import '../sink/sink.dart';
+import "byte_reader.dart";
+import "object_reader.dart";
+import "string_reader.dart";
+
+export "string_reader.dart" show StringSlice;
+
+/// A JSON reader which provides pull-based access to individual JSON tokens.
+///
+/// The JSON reader is intended to scan JSON source code from start to end.
```

Comment:
```suggestion
/// The JSON reader is intended to scan JSON source text from start to end.
```
(maybe, I imagine you're better at semantics)

Reply:
It's not bad. My only worry is that the byte-reader doesn't read *text*, but then, the object-reader doesn't read *code* either. So, accepted.